src/function/table/read_file.cpp

```diff
@@ -41,6 +74,11 @@ bool DirectMultiFileInfo<OP>::ParseOption(ClientContext &context, const string &
 template <class OP>
 unique_ptr<TableFunctionData> DirectMultiFileInfo<OP>::InitializeBindData(MultiFileBindData &multi_file_data,
                                                                           unique_ptr<BaseFileReaderOptions> options) {
+	if (multi_file_data.file_options.union_by_name) {
```

Comment:
We could remove the option instead:

```diff
diff --git a/src/function/table/read_file.cpp b/src/function/table/read_file.cpp
index d0481cc230..5ed232ab76 100644
--- a/src/function/table/read_file.cpp
+++ b/src/function/table/read_file.cpp
@@ -137,6 +137,7 @@ FileGlobInput DirectMultiFileInfo<OP>::GetGlobInput() {
 template <class OP>
 static TableFunction GetFunction() {
 	MultiFileFunction<DirectMultiFileInfo<OP>> table_function(OP::NAME);
+	table_function.named_parameters.erase("union_by_name");
 	return table_function;
 }
 ```

We may want to remove some others as well (see `MultiFileReader::AddParameters` for a list).